helix-core/src/main/java/org/apache/helix/util/RebalanceUtil.java

```diff
@@ -136,4 +141,15 @@
     result[1] = slaveStateValue;
     return result;
   }
+
+  public static void scheduleInstantPipeline(String clusterName) {
+    GenericHelixController controller = GenericHelixController.getController(clusterName);
+    if (controller != null) {
+      controller.scheduleInstantRebalance();
+    } else {
+      LOG.warn(
+          "Failed to issue a pipeline run for cluster {}. GenericHelixController has been constructed without any cluster name.",
```

Comment:
Okay, but this is a detail I think. When would GenericHelixController be created without a cluster name? How would that ever be possible?

If you want to take beef with the cluster name being null, then you should first validate the cluster name, correct?

Reply:
I added a check for the Cluster Name. I also tried to have better log message. Please have a look. Thanks :)

Reply:
It looks better now.